backend/src/graphql/resolver/TransactionResolver.ts

```diff
@@ -112,11 +114,41 @@ export class TransactionResolver {
     const self = new User(user)
     const transactions: Transaction[] = []
 
+    const transactionLinkRepository = getCustomRepository(TransactionLinkRepository)
+    const { sumHoldAvailableAmount, sumAmount } = await transactionLinkRepository.sumAmounts(
+      user.id,
+      now,
+    )
+
     // decay transaction
     if (!onlyCreations && currentPage === 1 && order === Order.DESC) {
       transactions.push(
         virtualDecayTransaction(lastTransaction.balance, lastTransaction.balanceDate, now, self),
       )
+      // open transaction link sum transaction
+      if (sumHoldAvailableAmount.greaterThan(0)) {
+        const lastTransactionLink = await dbTransactionLink.find({
+          where: { userId: self.id, redeemedBy: null, validUntil: MoreThan(now) },
+          order: { createdAt: 'DESC' },
+          take: 1,
+        })
+        const firstTransactionLink = await dbTransactionLink.find({
+          where: { userId: self.id, redeemedBy: null, validUntil: MoreThan(now) },
+          order: { createdAt: 'ASC' },
+          take: 1,
+        })
+        transactions.push(
+          virtualLinkTransaction(
+            lastTransaction.balance,
+            sumAmount,
+            sumHoldAvailableAmount,
+            sumHoldAvailableAmount.minus(sumAmount.toString()),
+            firstTransactionLink[0].createdAt,
```

Comment:
something named `singular` cant be an array.

What if given singular is not found?